internal/chacha/chacha.go

```diff
@@ -57,13 +57,6 @@ func setup(state *[64]byte, nonce, key []byte) (err error) {
 		HChaCha20(&tmpKey, &hNonce, &tmpKey)
 		copy(Nonce[8:], nonce[16:])
 		initialize(state, tmpKey[:], &Nonce)
-
-		// BUG(aead): A "good" compiler will remove this (optimizations)
-		//			  But using the provided key instead of tmpKey,
-		//			  will change the key (-> probably confuses users)
-		for i := range tmpKey {
-			tmpKey[i] = 0
-		}
```

Comment:
any reason to remove this? It probably isn't being optimized away, and zeroing keys is good practice.

Reply:
Well, yes, it wasn't optimized away.

And the way it's used within the package, it's a temporary key, derived from another temporary key, all stored in the stack.

This was the only key that's being zeroed, so it seemed redundant. But I'll revert this.

Depending on your willingness to upgrade to 1.21, we can use `clear`.

Reply:
I'm ambivalent; mostly I don't want to meddle with the original package too much, but maybe that ship has sailed. You're right that it's not very consistent to care about zeroing here and not in the rest of the package. (And frankly I've never been too concerned about key material floating around in program memory.)